### DISCUSSION TITLE
Don't allow setting "Default (EU)" controller region

### DISCUSSION THREAD

**Discussion by kpine** [zwave-js/zwave-js#7133] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?

Another user set their region to "Default (EU)" in ZUI's Z-Wave settings for a Zooz ZST39 (defaulting to US), believing it resulted in the controller being set to their region, EU (Europe).

This setting doesn't seem to be allowed by the controller, so it reverts to the previous. The user was unable to include or exclude devices until switching to the "Europe" region.

What did you expect to happen instead?

Driver should not allow setting this region, it seems to only exist it indicate a device is set to the default, which must be Europe.

  1. Set region to "Default (EU)"
  2. Notice in the driver logs the command fails

Anything else we should know?

Based on the public Simplicity SDK code, it appears the controller doesn't allow this setting, which aligns with the response received by the driver:

1. https://github.com/SiliconLabs/simplicity_sdk/blob/aa5ce2e835dfdce8c20fb828f27d3a261946f946/protocol/z-wave/Apps/zwave_ncp_serial_api/cmds_management.c#L240-L255
2. https://github.com/SiliconLabs/simplicity_sdk/blob/aa5ce2e835dfdce8c20fb828f27d3a261946f946/protocol/z-wave/ZAF/ApplicationUtilities/_commonIF/ZAF_Common_interface.c#L134-L154
3. https://github.com/SiliconLabs/simplicity_sdk/blob/aa5ce2e835dfdce8c20fb828f27d3a261946f946/protocol/z-wave/dist/include/zpal/zpal_radio.h#L327-L356

I did notice https://github.com/zwave-js/node-zwave-js/pull/7118 was implemented, which _might_ fix this for "newer" firmware? Not sure as that code doesn't look like it's public yet.

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs.log](https://github.com/user-attachments/files/16795301/zwavejs.log)

[a comment by an automated account is not shown]

**Comment by robertsLando** [comment, marked as answer]
@kpine Fixed on zui side: https://github.com/zwave-js/zwave-js-ui/commit/a9c8e3a3d14ebae3fc05bf7c9c290e1f627fdf0c
